packages/react-native/Libraries/AppDelegate/RCTReactNativeFactory.mm

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "RCTReactNativeFactory.h"
+#import <React/RCTColorSpaceUtils.h>
+#import <React/RCTLog.h>
+#import <React/RCTRootView.h>
+#import <React/RCTSurfacePresenterBridgeAdapter.h>
+#import <React/RCTUtils.h>
+#import <ReactCommon/RCTHost.h>
+#import <objc/runtime.h>
+#import <react/featureflags/ReactNativeFeatureFlags.h>
+#import <react/featureflags/ReactNativeFeatureFlagsDefaults.h>
+#import <react/renderer/graphics/ColorComponents.h>
+#import "RCTAppDelegate+Protected.h"
+#import "RCTAppSetupUtils.h"
+
+#if RN_DISABLE_OSS_PLUGIN_HEADER
+#import <RCTTurboModulePlugin/RCTTurboModulePlugin.h>
+#else
+#import <React/CoreModulesPlugins.h>
+#endif
+#import <React/RCTComponentViewFactory.h>
+#import <React/RCTComponentViewProtocol.h>
+#if USE_HERMES
+#import <ReactCommon/RCTHermesInstance.h>
+#else
+#import <ReactCommon/RCTJscInstance.h>
+#endif
+#import <react/nativemodule/defaults/DefaultTurboModules.h>
+
+#if __has_include(<ReactCodegen/RCTThirdPartyComponentsProvider.h>)
+#define USE_OSS_CODEGEN 1
+#import <ReactCodegen/RCTThirdPartyComponentsProvider.h>
+#else
+// Meta internal system do not generate the RCTModulesConformingToProtocolsProvider.h file
+#define USE_OSS_CODEGEN 0
+#endif
+
+using namespace facebook::react;
+
+@interface RCTReactNativeFactory () <RCTComponentViewFactoryComponentProvider, RCTHostDelegate>
+@end
+
+@implementation RCTReactNativeFactory
+
+- (instancetype)initWithDelegate:(id<RCTReactNativeFactoryDelegate>)delegate {
+  if (self = [super init]) {
+    [self _setUpFeatureFlags];
+    self.delegate = delegate;
+
+    RCTSetNewArchEnabled([self newArchEnabled]);
+    [RCTColorSpaceUtils applyDefaultColorSpace:[self defaultColorSpace]];
+    RCTEnableTurboModule([self turboModuleEnabled]);
+    
+    self.rootViewFactory = [self createRCTRootViewFactory];
+    
+    if ([self newArchEnabled] || [self fabricEnabled]) {
+      [RCTComponentViewFactory currentComponentViewFactory].thirdPartyFabricComponentsProvider = self;
+    }
+  }
+  
+  return self;
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
+                                                      jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
+{
+  return facebook::react::DefaultTurboModules::getTurboModule(name, jsInvoker);
+}
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
+                                                     initParams:
+                                                         (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+  return nullptr;
```

Comment:
This should check whether the delegate provide the method and use that. Otherwise return null.
How can we return pure c++ TM without that check, otherwise?

Reply:
From my research it looks like this was never called, created a separate PR to remove it: https://github.com/facebook/react-native/pull/48290